File: app/src/main/java/com/turlir/abakcalc/NotationRecyclerAdapter.java
package com.turlir.abakcalc;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class NotationRecyclerAdapter extends RecyclerView.Adapter<NotationRecyclerAdapter.Holder> {

    private final List<Object> mData;
    private final LayoutInflater mInflater;

    NotationRecyclerAdapter(Context cnt) {
        mData = new ArrayList<>();
        mInflater = LayoutInflater.from(cnt);
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = mInflater.inflate(R.layout.notation_item, parent, false);
        return new Holder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int position) {
        holder.bind(mData.get(position));
    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    <T> void setItems(List<T> items) {
        mData.clear();
        mData.addAll(items);
    }

    static class Holder extends RecyclerView.ViewHolder {

        Holder(View itemView) {
            super(itemView);
        }

        void bind(Object visual) {
            ((TextView) itemView).setText(visual.toString());
        }
    }

}

File: app/src/main/kotlin/com/turlir/abakcalc/NotationRecyclerAdapter.kt
package com.turlir.abakcalc

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import java.util.*

class NotationRecyclerAdapter constructor(cnt: Context) : RecyclerView.Adapter<NotationRecyclerAdapter.Holder>() {

    private val mData: MutableList<Any> = ArrayList()
    private val mInflater = LayoutInflater.from(cnt)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val v = mInflater.inflate(R.layout.notation_item, parent, false)
        return Holder(v)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        holder.bind(mData[position])
    }

    override fun getItemCount() = mData.size

    fun setItems(items: List<Any>) {
        mData.clear()
        mData.addAll(items)
    }

    class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        fun bind(visual: Any) {
            (itemView as TextView).text = visual.toString()
        }
    }

}
